### PR TITLE
Codecov coverage reports

### PR DESCRIPTION
**Pull request by jackreeceejini**
Testing Codecov coverage reports on a PR as per https://github.com/quantumlib/Cirq/issues/6063#issue-1667122621

### CODE REVIEW

**Comment by pavoljuhas** [comment]
Please rebase so this PR contains only codecov-related commits on top of the quantumlib/Cirq master.
As it is the PR has intermediate commits from your previous contribution.

One way of doing the rebase is
```
$ git checkout codecov-coverage-reports
$ git fetch https://github.com/quantumlib/Cirq.git master
$ git rebase -i FETCH_HEAD
  ... keep only the new commits
$ git push --force-with-lease your_remote_name codecov-coverage-reports
```

**Comment by jackreeceejini** [comment]
I will, thank you very much. I'm learning a lot!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -238,6 +238,11 @@ jobs:
         run: docker-compose -f cirq-rigetti/docker-compose.test.yaml up -d
       - name: Coverage check
         run: check/pytest-and-incremental-coverage -n auto --rigetti-integration
+      - name: Upload coverage reports to Codecov
+        uses: codecov/codecov-action@v3
+        env:
+          # Admin required to access repo settings > secrets and variable > actions
+          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```
Comment (pavoljuhas) [inline .github/workflows/ci.yml:245]: I think we can take this out, the token is not needed for public repositories - https://github.com/codecov/codecov-action#usage.

**Review by pavoljuhas — CHANGES_REQUESTED** [review]
@dstrain115 has installed the Codecov App and we got our first coverage report out at
https://app.codecov.io/github/quantumlib/Cirq/commit/1fd59723ba339b9281d150d31825c3791e4f0fdc/tree :tada:

LGTM after removing the unused token settings.

**Review by pavoljuhas — APPROVED** [review]
Thanks!

**Pull request merged**